File: core/java/src/main/java/hara/lang/protocol/INth.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.inth", name = "INth")
public interface INth<E> {
  @HaraMethod(value = "nth", arity = 2)
  E nth(long i);
}

File: core/rust/src/lang/protocol/inth.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.inth", name = "INth")]
pub trait INth<E> {
    #[hara_method(value = "nth", arity = 2, whole_wasm)]
    fn nth(&self, index: usize) -> Option<&E>;
}
